ihp-sg13g2/libs.tech/klayout/tech/drc/run_drc.py

```diff
@@ -0,0 +1,803 @@
+        return 0
+
+    if arguments.get("--density_only"):
+        run_check_by_flag("--density_only", "density")
+        check_drc_results(result_dbs, run_dir, layout_path, switches)
+        return 0
+
+    # Run main table check
+    table_arg = arguments.get("--table")
+    drc_file = generate_drc_run_template(rule_deck_full_path, run_dir, table_arg)
+    table_name = table_arg[0] if table_arg else "main"
+
+    result_dbs.append(run_check(drc_file, table_name, layout_path, run_dir, switches))
+
+    # Run additional checks if requested
+    if arguments.get("--antenna"):
+        run_check_by_flag("--antenna", "antenna")
+
+    if arguments.get("--density"):
+        run_check_by_flag("--density", "density")
+
+    if arguments.get("--MaxRuleSet"):
+        run_check_by_flag("--MaxRuleSet", "sg13g2_maximal")
+
+    # Final result verification
+    check_drc_results(result_dbs, run_dir, layout_path, switches)
+    return 0
+
+
+def main(run_dir: str, arguments: dict):
+    """
+    main function to run the DRC.
+
+    Parameters
+    ----------
+    run_dir : str
+        String with absolute path of the full run dir.
+    arguments : dict
+        Dictionary that holds the arguments used by user in the run command. This is generated by docopt library.
+    """
+
+    # Check gds file existance
+    if not os.path.exists(arguments["--path"]):
+        file_path = arguments["--path"]
+        logging.error(
+            f"The input GDS file path {file_path} doesn't exist, please recheck."
+        )
+        exit(1)
+
+    rule_deck_full_path = os.path.dirname(os.path.abspath(__file__))
+
+    # Check Klayout version
+    check_klayout_version()
+
+    # Check if there was a layout provided.
+    if not arguments["--path"]:
+        logging.error("No provided gds file, please add one")
+        exit(1)
+
+    # Check layout type
+    layout_path = arguments["--path"]
+    layout_path = check_layout_path(layout_path)
+
+    # Get run switches
+    switches = generate_klayout_switches(arguments, layout_path)
+
+    if workers_count == 1 or arguments["--antenna_only"] or arguments["--density_only"]:
+        run_single_processor(
+            arguments, rule_deck_full_path, layout_path, switches, run_dir
+        )
+    else:
+        run_parallel_run(arguments, rule_deck_full_path, layout_path, switches, run_dir)
+
+
+# ================================================================
+# -------------------------- MAIN --------------------------------
+# ================================================================
+
+if __name__ == "__main__":
+
+    # arguments
+    arguments = docopt(__doc__, version="RUN DRC: 1.0")
+
+    # run dir format
+    now_str = datetime.now(timezone.utc).strftime("drc_run_%Y_%m_%d_%H_%M_%S")
+
+    if (
+        arguments["--run_dir"] == "pwd"
+        or arguments["--run_dir"] == ""
+        or arguments["--run_dir"] is None
+    ):
+        run_dir = os.path.join(os.path.abspath(os.getcwd()), now_str)
```

Comment:
I would recommend using the `Pathlib`instead of `os.path`. Way more advanced!

Reply:
Will change that.

@FaragElsayed2

Reply:
Done

All `os.path` usages in the `run_drc.py` script have been replaced with `Pathlib`